### PR TITLE
Breaching charge

### PR DESCRIPTION
**Pull request by Diegoflores31**
## About The Pull Request

**Breaching Charge**
This Plastic explosive variant becomes a handy tool in quick assault operations featuring a faster planting time and smaller blast to quickly enter into fortified locations while keeping the operator safe.

After a fast planting this explosive will destroy the walls adjacent to it (half the damage as C4 and doesnt break reinforced walls) releasing deadly shrapnel at the opposite side of the wall causing damage to the enemies behind of it and providing a hole for the operators to start an assault.

**Limitations**
-cannot be planted on mobs
-cannot be planted on objects (including Burrower holes)
-no explosion damage/stun
-less damage than C4 to structures

**Pros**
-faster planting time
-faster explosion time
-shrapnel provides a good amount of damage to the target behind the wall
-safer since it uses a directional explosion

**Locations**
-Can be bought at Engie and SL vendors for 5 and 7 points
-Req Vendors will have some of them on stock each round
-Req can buy a crate of Breaching charges via ASRS

**concept video (final product may be a little different)**

https://user-images.githubusercontent.com/47069269/168969024-5c92e5b5-ba2a-42fa-a106-7c0bb00034fb.mp4

## Why It's Good For The Game

 Provides a new Tool for engineers in order to quickly breach into enemy structures with the added benefit of not killing the operators .
this is designed to compete with the C4 and Underbarrel shotgun to quickly break into the front.

## Changelog

:cl:
add: Adds Breaching Charge . A fast planting explosive that keeps the operator safe while emiting deadly shrapnel towards the target . This new explosive is available at Squad Engineer , SL and REQ vendors.
add: Adds Breaching Charge crate to ASRS

/:cl:

### CODE REVIEW

**Comment by realforest2001** [comment]
You'll need to update your changelog to the new format.

**Review by Agameofscones — CHANGES_REQUESTED** [review]
you've been cursed by html/lengord_pouches.yml. You need to rebase onto the latest copy of master and discard / delete that file. Otherwise the sprites look good to me.

**Review by Diegoflores31 — COMMENTED** [review]
file deleted . i guess im free from the curse now

**Review by Diegoflores31 — COMMENTED** [review]
.

**Comment by stanalbatross** [comment]
initial feedback so far seems to be that they take too long to detonate and are not useful enough. 

also fill out a more detailed changelog

**Comment by Katskan** [comment]
I never saw anyone using them. Possibly having a faster plant time would give them an edge, most people I saw were just talking about them as "worse C4"

**Comment by c4xmaniac** [comment]
I didn’t play much during the TM, but I never heard anyone using them and like Katskan said it seems to just be a worse C4. This being in combination with an intended master key nerf going in along side this MR, the plant time for the breaching charges should be decreased dramatically along with its point cost. I’m pretty sure I can buy c4 for the same price and it does basically the same thing with the addition of being able to blow tunnels.

**Comment by somepersondunno** [comment]
Make breaching charges cost 3 and regular Plastic Explosives cost 5 and this will be a lot more viable. I had a lot of fun playing engineer and using these to destroy walls and, as the name implies, BREACH into enemy defenses. Additionally, making them take a little less time to set up would be nice, but overall a lot of fun IMO.

**Comment by IngloriouslyBad** [comment]
Biggest problem with C4 and the new breaching charge is the set up times for both are just too long. Xenos aren't going to be surprised ever by C4 or breaching charges when you have to stand still ominously against a wall with a big red ring above your head, yelling C4 C4 C4. Xenos point at the wall before you're even done placing and get out of dodge or set up to counter you with the time it takes to set up and then go off. And good luck using either to widen a pre existing gap, a xeno will walk around the corner and push you 1 tile, stun you through the wall like you see with frontline burrowers, or just kill you in the case of T3s who turn the corner and tear you a new one, and retreat back inside.

**Comment by Sargeantmuffinman** [comment]
Just from arm chair thoughts, reading other comments and me testing this thing a couple of times in the T-Dome.
Too long to plant, xenos knowing exactly what's going to happen due to 2 red circles over a marine and cost.

Frankly if you addressed the planting time and the cost, it'd make it more worth while.
My two cents.
Cent 1: Planting time is 2 seconds.
Cent 2: Cost is either reduced or the breaching charge comes in stacks of 2 or 3 so you can plant multiple.

**Comment by Diegoflores31** [comment]
added faster planting time
added a new sharpnel type with a more appealing view than the standard
fixed some issues with the explosion eating a good amount of shrapnell
reduced amount of shrapnel but increased the damage
Adjusted Shrapnel spawn turf to avoid collitions with the wall

**Comment by Nanu308** [comment]
Balance lgtm, might need supply cost tweaks later down the line but we'll see.

File: code/game/objects/items/explosives/plastic.dm [hunk 1]
```diff
@@ -14,10 +14,13 @@
...
 	)
 
-	var/timer = 10
+	var/deploying_time = 50
+	var/penetration = 1 // How much damage adjacent walls receive
+	var/timer = 10 // detonation time
 	var/atom/plant_target = null //which atom the plstique explosive is planted on
 	var/overlay_image = "plastic-explosive2"
 	var/image/overlay
+	var/breachable = list(/obj/structure/window, /turf/closed, /obj/structure/machinery/door, /obj/structure/mineral_door)
```
Comment (Geevies) [inline code/game/objects/items/explosives/plastic.dm:23]:
```suggestion
	var/list/breachable = list(/obj/structure/window, /turf/closed, /obj/structure/machinery/door, /obj/structure/mineral_door)
```

File: code/datums/supply_packs/weapons.dm [hunk 2]
```diff
@@ -246,6 +246,17 @@
...
 	group = "Weapons"
 
+/datum/supply_packs/explosives_breaching_charge
+	name = "breaching charge crate (x4)"
+	contains = list(
+					/obj/item/explosive/plastic/breaching_charge,
+					/obj/item/explosive/plastic/breaching_charge,
+					/obj/item/explosive/plastic/breaching_charge,
+					/obj/item/explosive/plastic/breaching_charge)
+	cost = RO_PRICE_NORMAL
```
Comment (Geevies) [inline code/datums/supply_packs/weapons.dm:256]:
```suggestion
					/obj/item/explosive/plastic/breaching_charge
	)
	cost = RO_PRICE_NORMAL
```

**Review by Geevies — APPROVED** [review]

**Pull request merged**